src/main/scala/akka/stream/alpakka/xlsx/CellReference.scala

```diff
@@ -11,31 +11,32 @@ case class CellReference(name: String, colNum: Int, rowNum: Int)
 
 object CellReference {
 
+  private final val BaseLettersNum = 26
+  private final val AsciiCodeA = 65
+
   private val CELL_REF = """(\s+"""
 
   private def convertColStringToIndex(ref: String): Int = {
     @tailrec
-    def solver(refArray: List[Char], index: Int, retval: Int): Int = {
-      refArray.headOption match {
-        case Some(c) =>
-          if (index != 0) throw new IllegalArgumentException("Bad col ref format '" + ref + "'")
-          else solver(refArray.tail, index + 1, retval * 26 + c - 65 + 1)
+    def convert(refChars: List[Char], retval: Int): Int = {
+      refChars.headOption match {
+        case Some(char) =>
+          // `c - AsciiCodeA` converts base capital letters to consecutive numbers (starting from 0), ie. 'A' -> 0, 'B' -> 1, ...
+          convert(refChars.tail, retval * BaseLettersNum + (char - AsciiCodeA) + 1)
         case None => retval
       }
     }
 
-    solver(ref.toUpperCase(Locale.ROOT).toCharArray.toList, 0, 0)
+    convert(ref.toUpperCase(Locale.ROOT).toCharArray.toList, 0)
   }
 
   private def splitCellRef(ref: String) = {
-    val len  = ref.length
-    val s1   = ref.filterNot(c => c >= '0' && c <= '9')
-    val sLen = s1.length
-    if (len == 0 || sLen == len) {
-      None
-    } else {
-      Some((ref, s1, ref.substring(len - sLen)))
-    }
+    val refLength = ref.length
+    val refColumnPart       = ref.filterNot(c => c >= '0' && c <= '9')
+    val refColumnPartLength = refColumnPart.length
+
+    if (refLength == 0 || refColumnPartLength == refLength) None
+    else Some((ref, refColumnPart, ref.substring(refColumnPartLength)))
```

Comment:
is the library tested? If so, you could add 1 more test to check if that really works

Reply:
There are no tests yet - appropriate ticket is created and handled by our internal `gitlab` system.

Reply:
Does it mean you cannot test `parseRef` and `generateRef`? It's easier to do this incrementally on changed code than to write them all as a single task.